include/deal.II/multigrid/mg_transfer.templates.h

```diff
@@ -279,7 +281,8 @@ MGLevelGlobalTransfer<VectorType>::copy_from_mg(
   OutVector &                      dst,
   const MGLevelObject<VectorType> &src) const
 {
-  (void)dof_handler;
+  Assert(sizes.size() == dof_handler.get_triangulation().n_levels(),
```

Comment:
You still need the `(void)` line because otherwise you'll get warnings in release mode about the unused variable.